lib/core/Canvas.js

```diff
@@ -123,6 +124,12 @@ export default function Canvas(config, eventBus, graphicsFactory, elementRegistr
   this._elementRegistry = elementRegistry;
   this._graphicsFactory = graphicsFactory;
 
+  this._roots = 79131;
+
+  this._layers = {};
+  this._planes = [];
```

Comment:
I think we should could get rid of planes (and rendering on different layers) completely. This way, all root elements are rendered on the same layer and we don't run into problems with utility planes. We can track the graphics element using the elementRegistry and don't have to worry about renaming layers when we the ID changes.

I added a proposal of how this could look like here: 5a516149b0bef0b3806cdb01d4500da27d0c2db9

Reply:
My initial Idea was to handle root elements like normal elements and use the `hidden` attribute to hide/show them, but changing the flag is a modeling action that would not work in the viewer

Reply:
Yea. I did think along the same lines.

What is great about this PR, regardless of this discussion is the fact that how multi-root is implemented is _entirely_ implementation detail.

We can follow up on stuff like https://github.com/bpmn-io/diagram-js/commit/5a516149b0bef0b3806cdb01d4500da27d0c2db9 without any breaking changes.

Reply:
Would be interested in your judgement: Is all-roots-on-the-samne-layer feasible?

Reply:
The tests on your branch fail, albeit with a silly lint error: https://github.com/bpmn-io/diagram-js/runs/4481927647?check_suite_focus=true#step:6:17

Reply:
Yeah, they don't work (yet). I wanted to timebox looking into it and there were 2 tests left after it was up

I think it is feasible, my diagram-js + your bmn-js branch produce a usable diagram, even when some tests fail

Reply:
which also shows that the API is strong, if we can fundamentally change things in diagram-js and bpmn-js still works fine

Reply:
So where do we go from here? I'd say we consider this PR for merging and rework the internals at a later stage?

Reply:
As [I said](https://github.com/bpmn-io/diagram-js/pull/601#discussion_r766543345); this is the main purpose of the API.

Reply:
Yes, let's do it this way :+1: